packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts

```diff
@@ -0,0 +1,115 @@
+import { Undo } from 'roosterjs-editor-core';
+import { UndoSnapshotsService } from 'roosterjs-editor-core/lib/undo/UndoSnapshots';
+import { ImageManagerInteface, hasPlaceholder, UpdatePlaceholdersResult } from '../utils/ImageManager';
+
+// Max stack size that cannot be exceeded. When exceeded, old undo history will be dropped
+// to keep size under limit. This is kept at 10MB.
+const MAXSIZELIMIT = 10000000;
+
+export default class UndoWithImagePlugin extends Undo {
+    /**
+     * Create an instance of Undo
+     * @param preserveSnapshots True to preserve the snapshots after dispose, this allows
+     * this object to be reused when editor is disposed and created again
+     * @param bufferSize The buffer size for snapshots. Default value is 10MB, it is possible after
+     * placeholder to image resolution that buffer size is greater.
+     */
+    constructor(
+        private imageManager: ImageManagerInteface,
+        preserveSnapshots?: boolean,
+        private bufferSize: number = MAXSIZELIMIT
+    ) {
+        super(preserveSnapshots, bufferSize);
+    }
+
+    /**
```

Comment:
nit: not needed?

Reply:
I mean the whole dispose function here (as usual GitHub doesn't preserve highlighting :(

Reply:
deadcode